source/distributed/tria.cc

```diff
@@ -2219,7 +2204,7 @@ namespace parallel
       Assert (refinement_in_progress == false, ExcInternalError());
 
       // get rid of the unique communicator used here again
-      MPI_Comm_free (&mpi_communicator);
+      MPI_Comm_free (&this->mpi_communicator);
```

Comment:
This actually needs to move to the base class. We used to duplicate the communicator in the constructor above, but now you only pass it down to the base class. If the base class duplicates the communicator, then it also needs to free it.